tests/test_iot/test_iot.py

```diff
@@ -756,6 +756,140 @@ def test_delete_principal_thing():
+    #        |--3d
+    # --1b
+
+    # create thing groups tree
+    # 1
+    thing_group1a = client.create_thing_group(thingGroupName=group_name_1a)
+    thing_group1a.should.have.key("thingGroupName").which.should.equal(group_name_1a)
+    thing_group1a.should.have.key("thingGroupArn")
+    thing_group1b = client.create_thing_group(thingGroupName=group_name_1b)
+    thing_group1b.should.have.key("thingGroupName").which.should.equal(group_name_1b)
+    thing_group1b.should.have.key("thingGroupArn")
+    # 2
+    thing_group2a = client.create_thing_group(
+        thingGroupName=group_name_2a, parentGroupName=group_name_1a
+    )
+    thing_group2a.should.have.key("thingGroupName").which.should.equal(group_name_2a)
+    thing_group2a.should.have.key("thingGroupArn")
+    thing_group2b = client.create_thing_group(
+        thingGroupName=group_name_2b, parentGroupName=group_name_1a
+    )
+    thing_group2b.should.have.key("thingGroupName").which.should.equal(group_name_2b)
+    thing_group2b.should.have.key("thingGroupArn")
+    # 3
+    thing_group3a = client.create_thing_group(
+        thingGroupName=group_name_3a, parentGroupName=group_name_2a
+    )
+    thing_group3a.should.have.key("thingGroupName").which.should.equal(group_name_3a)
+    thing_group3a.should.have.key("thingGroupArn")
+    thing_group3b = client.create_thing_group(
+        thingGroupName=group_name_3b, parentGroupName=group_name_2a
+    )
+    thing_group3b.should.have.key("thingGroupName").which.should.equal(group_name_3b)
+    thing_group3b.should.have.key("thingGroupArn")
+    thing_group3c = client.create_thing_group(
+        thingGroupName=group_name_3c, parentGroupName=group_name_2b
+    )
+    thing_group3c.should.have.key("thingGroupName").which.should.equal(group_name_3c)
+    thing_group3c.should.have.key("thingGroupArn")
+    thing_group3d = client.create_thing_group(
+        thingGroupName=group_name_3d, parentGroupName=group_name_2b
+    )
+    thing_group3d.should.have.key("thingGroupName").which.should.equal(group_name_3d)
+    thing_group3d.should.have.key("thingGroupArn")
+
+    # begin tests
+    # should list all groups
+    resp = client.list_thing_groups()
+    resp.should.have.key("thingGroups")
+    resp["thingGroups"].should.have.length_of(8)
+    # should list all groups non-recursively
+    resp = client.list_thing_groups(recursive=False)
+    resp.should.have.key("thingGroups")
+    resp["thingGroups"].should.have.length_of(2)
+
+    # should list all groups filtered by parent
+    resp = client.list_thing_groups(parentGroup=group_name_1a)
+    resp.should.have.key("thingGroups")
+    resp["thingGroups"].should.have.length_of(6)
+    resp = client.list_thing_groups(parentGroup=group_name_2a)
+    resp.should.have.key("thingGroups")
+    resp["thingGroups"].should.have.length_of(2)
+    resp = client.list_thing_groups(parentGroup=group_name_1b)
+    resp.should.have.key("thingGroups")
+    resp["thingGroups"].should.have.length_of(0)
+    try:
```

Comment:
Please use the `with assert_raises(ClientException) as e`, here, similar to https://github.com/spulec/moto/blob/111d0519f5bcde1fa915c3f3aefbae1c70b6c745/tests/test_config/test_config.py#L31

Reply:
Done